core/src/main/clojure/xtdb/trie.clj

```diff
@@ -310,6 +312,9 @@
 
           {l0-trie-keys 0, l1-trie-keys 1, :as level-grouped-file-names} (file-names->level-groups file-names)
 
+          ;; filtering superseded L1 files
+          l1-trie-keys (->> l1-trie-keys (sort-by (juxt :first-row :next-row) #(compare %2 %1)) (util/distinct-by :first-row) reverse)
```

Comment:
We can cut down on some of the more expensive operations and intermediate data structures here because the file names are already (and still, after this PR :clap:) lexicographically sorted - the files with the same first row should (IIRC) already be next to each other in `l1-trie-keys`.

i.e. you should then be able to `(partition-by :first-row)` rather than sorting and distincting, `(map last)` and possibly save yourself the `reverse` too - `(into [] (comp (partition-by :first-row) (map last)) l1-trie-keys)`?

Reply:
Hadn't realized that we have this guarantee on the buffer pool.